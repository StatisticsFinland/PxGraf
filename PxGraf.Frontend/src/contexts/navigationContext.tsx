import * as React from "react";

interface INavigationContextData {
    tablePath: string[] | null,
}

interface INavigationContextMethods {
    setTablePath: React.Dispatch<React.SetStateAction<string[] | null>>,
}

type TNavigationContextValue = INavigationContextData & INavigationContextMethods;

interface INavigationProviderProps {
    children: React.ReactNode;
}

const NavigationContext = React.createContext<(TNavigationContextValue)>(null);

export const NavigationProvider: React.FC<INavigationProviderProps> = ({ children }) => {
    const [tablePath, setTablePath] = React.useState<string[] | null>(null);

    const contextValue: TNavigationContextValue = React.useMemo(() => ({
        tablePath,
        setTablePath,
    }), [tablePath]);

    return <NavigationContext.Provider value={contextValue}>{children}</NavigationContext.Provider>
}

export const useNavigationContext = (): TNavigationContextValue => {
    return React.useContext(NavigationContext);
}
