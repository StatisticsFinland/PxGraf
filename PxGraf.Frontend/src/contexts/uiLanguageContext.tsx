/* istanbul ignore file */

import * as React from 'react';
import { useTranslation } from 'react-i18next';

/**
 * Interface for UI language context properties
 * @property language - The current UI language. These languages are defined in ../localization/translationsConfig.json.
 * @property setLanguage - Function to set the UI language
 * @property languageTab - The current content language. These languages are defined by the content retrieved from the API and are used to customize the visualization.
 * @property setLanguageTab - Function to set the content language
 * @property availableUiLanguages - List of available UI languages
 * @property uiContentLanguage - The current UI content language. These languages are defined by the content retrieved from the API, but are used in the UI.
 * @property setUiContentLanguage - Function to set the UI content language
 */
interface IUiLanguageContext {
    language: string;
    setLanguage: React.Dispatch<string>;
    languageTab: string;
    setLanguageTab: React.Dispatch<string>;
    availableUiLanguages: string[];
    uiContentLanguage: string;
    setUiContentLanguage: React.Dispatch<string>;
}

export const UiLanguageContext = React.createContext<IUiLanguageContext>({
    language: "",
    setLanguage: () => { /* no base implementation */ },
    languageTab: '',
    setLanguageTab: () => { /* no base implementation */ },
    availableUiLanguages: [],
    uiContentLanguage: '',
    setUiContentLanguage: () => { /* no base implementation */ }
});

/**
 * Provider for the UI language context
 * @param children - The child components
 */
export const UiLanguageProvider: React.FC<{ children: React.ReactNode }> = ({ children }) => {
    const { i18n } = useTranslation();
    const [language, setLanguage] = React.useState(i18n.language);
    const [languageTab, setLanguageTab] = React.useState(i18n.language);
    const [availableUiLanguages, setAvailableUiLanguages] = React.useState<string[]>([]);
    const [uiContentLanguage, setUiContentLanguage] = React.useState(i18n.language);

    const loadLanguages = async () => {
        // Load available languages from i18n
        const languages = Object.keys(i18n.services.resourceStore.data);
        setAvailableUiLanguages(languages);
        const defaultLanguage = languages.includes(i18n.language) ? i18n.language : languages[0];
        if (language !== defaultLanguage) {
            setLanguage(defaultLanguage);
        }
    }

    React.useEffect(() => {
        loadLanguages();
    }, []);

    const setUiLang = (lang: string) => {
        i18n.changeLanguage(lang);
        setLanguage(lang);
    }

    const contextValue = React.useMemo(() => {
        return { language, setLanguage: setUiLang, languageTab, setLanguageTab, availableUiLanguages, uiContentLanguage, setUiContentLanguage }
    }, [language, languageTab, availableUiLanguages, uiContentLanguage]);

    return (
        <UiLanguageContext.Provider value={contextValue}>
            {children}
        </UiLanguageContext.Provider>
    );
}

export default UiLanguageContext;
