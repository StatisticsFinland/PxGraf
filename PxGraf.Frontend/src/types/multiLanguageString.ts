/* istanbul ignore file */

/**
 * Type for string supporting multiple languages with each key representing a language and the value the string in that language.
 */
export type MultiLanguageString = { [key: string]: string };