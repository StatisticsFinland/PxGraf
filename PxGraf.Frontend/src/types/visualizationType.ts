/* istanbul ignore file */

/**
 * Enum that defines different visualization types.
 */
export enum VisualizationType {
    VerticalBarChart = 'VerticalBarChart',
    GroupVerticalBarChart = 'GroupVerticalBarChart',
    StackedVerticalBarChart = 'StackedVerticalBarChart',
    PercentVerticalBarChart = 'PercentVerticalBarChart',
    HorizontalBarChart = 'HorizontalBarChart',
    GroupHorizontalBarChart = 'GroupHorizontalBarChart',
    StackedHorizontalBarChart = 'StackedHorizontalBarChart',
    PercentHorizontalBarChart = 'PercentHorizontalBarChart',
    PyramidChart = 'PyramidChart',
    PieChart = 'PieChart',
    LineChart = 'LineChart',
    ScatterPlot = 'ScatterPlot',
    Table = 'Table'
}