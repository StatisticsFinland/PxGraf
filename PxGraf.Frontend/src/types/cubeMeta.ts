/* istanbul ignore file */

import { MultiLanguageString } from "./multiLanguageString";

/**
 * Interface for cube meta properties. A cube is a multi-dimensional result of Px file data filtered by a query.
 * @property {string[]} languages - List of available languages.
 * @property {MultiLanguageString} header - The multi language header of the cube.
 * @property {MultiLanguageString} note - The note of the cube.
 * @property {IVariable[]} variables - List of variables.
 */
export interface IMatrixMetadata {
    DefaultLanguage: string,
    AvailableLanguages: string[],
    Dimensions: IDimension[],
    AdditionalProperties?: { [key: string]: IMetaProperty }
}

/**
 * TODO: Summary
 */
export interface IMetaProperty {
    KeyWord: string,
    CanGetStringValue: boolean,
    CanGetMultilanguageValue: boolean,
    Entries: MultiLanguageString | string
}

/**
 * Interface for dimnension properties. Each variable defines a dimension in a cube.
 * @property {string} code - The code of the variable.
 * @property {VariableType} type - The type of the variable.
 * @property {MultiLanguageString} name - The multi language name of the variable.
 * @property {{ [key: string]: IMetaProperty }} additionalProperties - Optional dictionary for additional metadata properties
 * @property {IVariableValue[]} values - List of values associated with this variable.
 */
export interface IDimension {
    Code: string,
    Name: MultiLanguageString,
    AdditionalProperties?: { [key: string]: IMetaProperty },
    Values: IDimensionValue[]
    Type: VariableType,
}

/**
 * Interface for variable properties.Each variable defines a dimension in a cube.
 * @property { string } code - The code of the variable.
 * @property { MultiLanguageString } name - The multi language name of the variable.
 * @property { MultiLanguageString } note - The multi language note of the variable.
 * @property { VariableType } type - The type of the variable.
 * @property { IVariableValue[] } values - List of values associated with this variable.
 */
export interface IVariable {
    code: string,
    name: MultiLanguageString,
    note: MultiLanguageString,
    type: VariableType,
    values: IVariableValue[]
}

/**
 * Interface for dimension value properties.
 * @property {string} code - The code of the value.
 * @property {MultiLanguageString} name - The multi language name of the value.
 * @property {{ [key: string]: IMetaProperty }} additionalProperties - Optional dictionary for additional metadata properties
 */
export interface IDimensionValue {
    Code: string,
    Name: MultiLanguageString,
    Virtual: boolean
    AdditionalProperties?: { [key: string]: IMetaProperty },
}

/**
 * Interface for variable value properties.
 * @property { string } code - The code of the value.
 * @property { MultiLanguageString } name - The multi language name of the value.
 * @property { MultiLanguageString } note - The multi language note of the value.
 * @property { boolean } isSum - Flag to indicate if the value is an elimination value.
 * @property { IContentComponent } contentComponent - The content variable component of the value.
 * @property { IVirtualComponent[] } virtualComponents - List of virtual components.
 */
export interface IVariableValue {
    code: string,
    name: MultiLanguageString,
    note: MultiLanguageString,
    isSum: boolean,
    contentComponent?: IContentComponent
    virtualComponents?: IVirtualComponent[]
}

/**
 * Interface for value content variable value properties.
 * @property {MultiLanguageString} unit - The unit of the value.
 * @property {MultiLanguageString} source - The source of the value.
 * @property {number} numberOfDecimals - The precision, number of decimals of the value.
 * @property {string} lastUpdated - The last updated date of the value.
 */
export interface IContentComponent {
    unit: MultiLanguageString,
    source: MultiLanguageString,
    numberOfDecimals: number,
    lastUpdated: string
}

/**
 * Interface for virtual component properties.
 * @property {string} operator - The operator of the virtual component.
 * @property {string[]} operandCoded - List of coded operands.
 */
export interface IVirtualComponent {
    operator: string,
    operandCoded: string[]
}

/**
 * Enumeration for variable types.
 */
export enum VariableType {
    Unknown = 'N',
    Time = 'T',
    Ordinal = 'P',
    Geographical = 'G',
    Content = 'C',
    OtherClassificatory = 'F',
}

export function getVariablesFromMatrixMetadata(meta: IMatrixMetadata): IVariable[] {
    return meta.Dimensions.map(d => {
        return {
            code: d.Code,
            name: d.Name,
            note: d.Name,
            type: d.Type,
            values: d.Values.map(v => convertDimensionValueToVariableValue(v, meta))
        }
    })
}

export function convertDimensionValueToVariableValue(value: IDimensionValue, meta: IMatrixMetadata): IVariableValue {
    const dimension: IDimension = meta.Dimensions.find(d => d.Values.find(v => v.Code === value.Code));
    const contentComponent = (): IContentComponent | null => {
        if (dimension.Type === VariableType.Content) {
            return {
                unit: getAdditionalProperty("UNIT", value.AdditionalProperties) as MultiLanguageString,
                source: getAdditionalProperty("SOURCE", value.AdditionalProperties) as MultiLanguageString,
                numberOfDecimals: parseInt(getAdditionalProperty("PRECISION", value.AdditionalProperties, true) as string),
                lastUpdated: getAdditionalProperty("LAST-UPDATED", value.AdditionalProperties, true) as string
            }
        }
        else return null;
    }
    return {
        code: value.Code,
        name: value.Name,
        note: value.Name,
        isSum: getValueIsSumValue(value, dimension),
        contentComponent: contentComponent()
    }
}

export function getValueIsSumValue(value: IDimensionValue, dimension: IDimension): boolean {
    return getAdditionalProperty("ELIMINATION", dimension.AdditionalProperties) === value.Code;
}

function getAdditionalProperty(key: string, additionalProperties: { [key: string]: IMetaProperty }, forceToString = false): string | MultiLanguageString {
    if (additionalProperties[key]) {
        if (additionalProperties[key].CanGetStringValue) {
            return additionalProperties[key].Entries[0];
        }
        else if (additionalProperties[key].CanGetMultilanguageValue) {
            if (forceToString) return additionalProperties[key].Entries[0];
            else return additionalProperties[key].Entries;
        }
    }
    return null;
}
