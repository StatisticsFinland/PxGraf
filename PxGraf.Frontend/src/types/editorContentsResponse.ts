import { MultiLanguageString } from "./multiLanguageString";
import { VisualizationType } from "./visualizationType";

/**
 * Interface for dimension sorting options.
 * @property {string} code - The code of the sorting option.
 * @property {MultiLanguageString} description - Longer, user facing description of the sorting option.
 */
export interface ISortingOption {
    code: string;
    description: MultiLanguageString;
}

/**
 * Interface for sorting option collection. Contains default and pivoted sorting options if available.
 * @property {ISortingOption[]} default - Default sorting options.
 * @property {ISortingOption[]} pivoted - Pivoted sorting options.
 */
interface ISortingOptionCollection {
    default: ISortingOption[];
    pivoted: ISortingOption[];
}

/**
 * Interface for visualization options for a specific visualization type.
 * @property {VisualizationType} type - The type of visualization.
 * @property {boolean} [allowShowingDataPoints] - Flag to indicate if showing data points is allowed.
 * @property {boolean} [allowCuttingYAxis] - Flag to indicate if cutting the Y-axis is allowed.
 * @property {boolean} [allowMatchXLabelsToEnd] - Flag to indicate if matching X labels to the end is allowed.
 * @property {boolean} [allowSetMarkerScale] - Flag to indicate if setting marker scale is allowed.
 * @property {boolean} allowManualPivot - Flag to indicate if manual pivoting is allowed.
 * @property {boolean} allowMultiselect - Flag to indicate if multivalue selectable is allowed.
 * @property {ISortingOptionCollection} sortingOptions - Sorting options for the visualization containing default and pivoted options.
 */
export interface IVisualizationOptions {
    type: VisualizationType;
    allowShowingDataPoints?: boolean;
    allowCuttingYAxis?: boolean;
    allowMatchXLabelsToEnd?: boolean;
    allowSetMarkerScale?: boolean;
    allowManualPivot: boolean;
    allowMultiselect: boolean;
    sortingOptions: ISortingOptionCollection;
}

/**
 * Interface for the response from the editor contents API used to populate the settings for adjusting the visualization in the editor.
 * @property {number} size - The size of the response based on the amount of values selected from all dimensions.
 * @property {number} sizeWarningLimit - The size warning limit for the response.
 * @property {number} maximumSupportedSize - The maximum supported size for the response.
 * @property {MultiLanguageString} headerText - Default header text for the visualization as a @ {@link MultiLanguageString} object.
 * @property {number} maximumHeaderLength - The maximum length of the header text.
 * @property {IVisualizationOptions[]} visualizationOptions - Visualization options for the valid visualization types as an array of @ {@link IVisualizationOptions} objects.
 * @property {object} visualizationRejectionReasons - Object containing rejection reasons for each visualization type as a @ {@link MultiLanguageString} object.
 */
export interface IEditorContentsResponse {
    size: number;
    sizeWarningLimit: number;
    maximumSupportedSize: number;
    headerText: MultiLanguageString;
    maximumHeaderLength: number;
    visualizationOptions: IVisualizationOptions[];
    visualizationRejectionReasons: {
        [visualizationType: string]: MultiLanguageString;
    }
}