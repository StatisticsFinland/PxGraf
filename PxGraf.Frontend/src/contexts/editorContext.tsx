/* istanbul ignore file */

import { debounce } from 'lodash';
import * as React from 'react';
import { ICubeQuery, Query } from 'types/query';
import { IVisualizationSettings } from 'types/visualizationSettings';
import { VisualizationType } from 'types/visualizationType';

/**
 * Interface for editor context properties
 * @property {ICubeQuery} cubeQuery - Query object that contains the header and dimension queries
 * @property {React.Dispatch<ICubeQuery>} setCubeQuery - Function to set the cube query
 * @property {Query} query - Query object that contains the dimension queries
 * @property {React.Dispatch<Query>} setQuery - Function to set the query
 * @property {boolean} saveDialogOpen - Flag to indicate if the save dialog is open
 * @property {React.Dispatch<boolean>} setSaveDialogOpen - Function to set the save dialog open flag
 * @property {VisualizationType} selectedVisualizationUserInput - The selected visualization type
 * @property {React.Dispatch<VisualizationType>} setSelectedVisualizationUserInput - Function to set the selected visualization type
 * @property {IVisualizationSettings} visualizationSettingsUserInput - The visualization settings
 * @property {React.Dispatch<IVisualizationSettings>} setVisualizationSettingsUserInput - Function to set the visualization settings
 * @property {{ [key: string]: string[] }} defaultSelectables - The default selectables for each dimension
 * @property {React.Dispatch<{ [key: string]: string[] }>} setDefaultSelectables - Function to set the default selectables
 */
interface IEditorContext {
    cubeQuery: ICubeQuery;
    setCubeQuery: React.Dispatch<ICubeQuery>;
    query: Query;
    setQuery: React.Dispatch<Query>;
    saveDialogOpen: boolean;
    setSaveDialogOpen: React.Dispatch<boolean>;
    selectedVisualizationUserInput: VisualizationType;
    setSelectedVisualizationUserInput: React.Dispatch<VisualizationType>;
    visualizationSettingsUserInput: IVisualizationSettings;
    setVisualizationSettingsUserInput: React.Dispatch<IVisualizationSettings>;
    defaultSelectables: { [key: string]: string[] };
    setDefaultSelectables: React.Dispatch<{ [key: string]: string[] }>
}

/**
 * Context required for the editor
 */
export const EditorContext = React.createContext<IEditorContext>({
    cubeQuery: null,
    setCubeQuery: () => { /* no base implementation */ },
    query: null,
    setQuery: () => { /* no base implementation */ },
    saveDialogOpen: false,
    setSaveDialogOpen: () => { /* no base implementation */ },
    selectedVisualizationUserInput: null,
    setSelectedVisualizationUserInput: () => { /* no base implementation */ },
    visualizationSettingsUserInput: null,
    setVisualizationSettingsUserInput: () => { /* no base implementation */ },
    defaultSelectables: null,
    setDefaultSelectables: () => { /* no base implementation */ }
});

/**
 * Provider for the editor context
 * @param children - The child components
 */
export const EditorProvider: React.FC<{ children: React.ReactNode }> = ({ children }) => {
    const [cubeQuery, setCubeQueryState] = React.useState({ variableQueries: {} });
    const [query, setQuery] = React.useState(null);
    const [saveDialogOpen, setSaveDialogOpen] = React.useState(false);
    const [selectedVisualizationUserInput, setSelectedVisualizationUserInput] = React.useState(null);
    const [visualizationSettingsUserInput, setVisualizationSettingsUserInput] = React.useState(null);
    const [defaultSelectables, setDefaultSelectables] = React.useState(null);

    const debouncedSetCubeQuery = React.useMemo(() => {
        return debounce((newQuery: ICubeQuery) => {
            setCubeQueryState(newQuery);
        }, 1000);
    }, []);

    const setCubeQuery = React.useCallback((newQuery: ICubeQuery) => {
        debouncedSetCubeQuery(newQuery);
    }, [debouncedSetCubeQuery]);

    React.useEffect(() => {
        return () => {
            debouncedSetCubeQuery.cancel();
        };
    }, [debouncedSetCubeQuery]);

    const contextValue = React.useMemo(() => {
        return {
            cubeQuery, setCubeQuery,
            query, setQuery,
            saveDialogOpen, setSaveDialogOpen,
            selectedVisualizationUserInput, setSelectedVisualizationUserInput,
            visualizationSettingsUserInput, setVisualizationSettingsUserInput,
            defaultSelectables, setDefaultSelectables
        }
    }, [
        cubeQuery, query, saveDialogOpen, selectedVisualizationUserInput,
        visualizationSettingsUserInput, defaultSelectables
    ]);

    return (
        <EditorContext.Provider value={contextValue}>
            {children}
        </EditorContext.Provider>
    );
}