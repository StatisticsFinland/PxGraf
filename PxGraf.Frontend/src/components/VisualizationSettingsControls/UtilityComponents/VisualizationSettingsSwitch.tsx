import React from 'react';
import { useTranslation } from 'react-i18next';
import { FormControlLabel, FormControl, Switch, Typography } from '@mui/material';
import { IVisualizationSettings } from "types/visualizationSettings";
import styled from "styled-components";
import { EditorContext } from '../../../contexts/editorContext';

const HiddenFormControl = styled(FormControl)`
    display: none !important;
`;

const LabelTypography = styled(Typography)`
  min-width: 125px;
  max-width: 200px;
`;

interface IVisualizationSettingsSwitchProps {
    selected: boolean,
    visualizationSettings: IVisualizationSettings,
    label: string,
    changeProperty: keyof IVisualizationSettings,
    hidden?: boolean
}

export const VisualizationSettingsSwitch: React.FC<IVisualizationSettingsSwitchProps> = ({ selected, visualizationSettings, label, changeProperty, hidden }) => {
    const { t } = useTranslation();
    const Control = hidden ? HiddenFormControl : FormControl;
    const { setVisualizationSettingsUserInput } = React.useContext(EditorContext);

    return (
        <Control fullWidth>
            <FormControlLabel
                control={
                    <Switch
                        checked={selected}
                        onChange={(event) => setVisualizationSettingsUserInput({ ...visualizationSettings, [changeProperty]: event.target.checked })}
                    />
                }
                label={<LabelTypography>{t(label)}</LabelTypography>}
            />
        </Control>
    )
}

export default VisualizationSettingsSwitch;