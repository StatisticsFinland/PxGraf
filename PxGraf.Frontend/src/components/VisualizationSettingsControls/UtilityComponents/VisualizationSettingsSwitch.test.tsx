import React from 'react';
import { fireEvent, render } from "@testing-library/react";
import VisualizationSettingsSwitch from "./VisualizationSettingsSwitch";
import { EditorContext } from '../../../contexts/editorContext';
import { VisualizationType } from '../../../types/visualizationType';

jest.mock('react-i18next', () => ({
    ...jest.requireActual('react-i18next'),
    useTranslation: () => {
        return {
            t: (str: string) => str,
            i18n: {
                changeLanguage: () => new Promise(() => null),
            },
        };
    },
}));

const mockSettingsChangedHandler = jest.fn();
const mockVisualizationSettings = { showDataPoints: false };

describe('Rendering test', () => {
    it('renders correctly', () => {
        const { asFragment } = render(
            <VisualizationSettingsSwitch
                selected={true}
                visualizationSettings={mockVisualizationSettings}
                label={"label"}
                changeProperty={"showDataPoints"}
            />);
        expect(asFragment()).toMatchSnapshot();
    });
});

describe('Assertion tests', () => {
    it('calls settingsChangedHandler with correct value when clicked', () => {
        const { getByRole } = render(
            <EditorContext.Provider value={{
                defaultSelectables: {},
                setDefaultSelectables: jest.fn(),
                cubeQuery: null,
                setCubeQuery: jest.fn(),
                query: {},
                setQuery: jest.fn(),
                saveDialogOpen: false,
                setSaveDialogOpen: jest.fn(),
                selectedVisualizationUserInput: VisualizationType.VerticalBarChart,
                setSelectedVisualizationUserInput: jest.fn(),
                visualizationSettingsUserInput: {},
                setVisualizationSettingsUserInput: mockSettingsChangedHandler
            }}>
            <VisualizationSettingsSwitch
                selected={false}
                visualizationSettings={mockVisualizationSettings}
                label={"label"}
                changeProperty={"showDataPoints"}
                />
            </EditorContext.Provider>);
        const switchElement = getByRole('checkbox');
        fireEvent.click(switchElement);

        expect(mockSettingsChangedHandler).toHaveBeenCalledWith({ ...mockVisualizationSettings, showDataPoints: true });
    });
});