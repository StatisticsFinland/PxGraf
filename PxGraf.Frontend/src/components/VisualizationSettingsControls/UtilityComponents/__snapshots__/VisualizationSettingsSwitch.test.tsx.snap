// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Rendering test renders correctly 1`] = `
<DocumentFragment>
  <div
    class="MuiFormControl-root MuiFormControl-fullWidth css-ytlejw-MuiFormControl-root"
  >
    <label
      class="MuiFormControlLabel-root MuiFormControlLabel-labelPlacementEnd css-19opgx6-MuiFormControlLabel-root"
    >
      <span
        class="MuiSwitch-root MuiSwitch-sizeMedium css-rehya9-MuiSwitch-root"
      >
        <span
          class="MuiButtonBase-root MuiSwitch-switchBase MuiSwitch-colorPrimary Mui-checked PrivateSwitchBase-root MuiSwitch-switchBase MuiSwitch-colorPrimary Mui-checked Mui-checked css-161ms7l-MuiButtonBase-root-MuiSwitch-switchBase"
        >
          <input
            checked=""
            class="PrivateSwitchBase-input MuiSwitch-input css-j8yymo"
            type="checkbox"
          />
          <span
            class="MuiSwitch-thumb css-17jyosd-MuiSwitch-thumb"
          />
        </span>
        <span
          class="MuiSwitch-track css-tj8d99-MuiSwitch-track"
        />
      </span>
      <span
        class="MuiTypography-root MuiTypography-body1 MuiFormControlLabel-label css-rizt0-MuiTypography-root"
      >
        <p
          class="MuiTypography-root MuiTypography-body1 sc-fYsHOw ditWhj css-rizt0-MuiTypography-root"
        >
          label
        </p>
      </span>
    </label>
  </div>
</DocumentFragment>
`;
